### PR TITLE
Modernize transforms tutorial to torchvision v2 API

### LINKED ISSUE #3853: Update Transforms tutorial — replace deprecated ToTensor(), migrate to v2 namespace, modernize one-hot encoding
## Description

The [Transforms](https://docs.pytorch.org/tutorials/beginner/basics/transforms_tutorial.html) tutorial uses the deprecated `torchvision.transforms.ToTensor()` transform, operates entirely in the legacy v1 namespace, and uses a low-level `scatter_` pattern for one-hot encoding where `F.one_hot` is the modern alternative.

###  Deprecated APIs

| Issue | Current Code | Replacement | Since |
|-------|-------------|-------------|-------|
| `torchvision.transforms.ToTensor` | `from torchvision.transforms import ToTensor, Lambda` / `transform=ToTensor()` (lines 31, 37) | `from torchvision.transforms import v2` / `transform=v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)])` | torchvision 0.16 (Oct 2023) — `ToTensor` emits a deprecation warning; v2 transforms are the supported API |

### Suboptimal / Outdated Patterns

| Issue | Current Code | Modern Alternative | Notes |
|-------|-------------|-------------------|-------|
| v1 `transforms` namespace throughout | `from torchvision.transforms import ToTensor, Lambda` | `from torchvision.transforms import v2` | The entire tutorial operates in the legacy namespace. All imports and explanations should shift to v2. |
| `Lambda` for one-hot encoding | `Lambda(lambda y: torch.zeros(10, dtype=torch.float).scatter_(0, torch.tensor(y), value=1))` (line 38) | `torch.nn.functional.one_hot(torch.tensor(y), num_classes=10).float()` | `F.one_hot` is cleaner and more readable. `Lambda` itself still works in v2, but the one-hot pattern has a better built-in. |
| `scatter_` for one-hot encoding | `torch.zeros(10, dtype=torch.float).scatter_(dim=0, index=torch.tensor(y), value=1)` (lines 60-61) | `torch.nn.functional.one_hot(torch.tensor(y), num_classes=10).float()` | `scatter_` is a low-level op; `F.one_hot` is purpose-built and self-documenting. |

### Specific lines

- **Line 31**: `from torchvision.transforms import ToTensor, Lambda` -> `from torchvision.transforms import v2`
- **Line 37**: `transform=ToTensor(),` -> `transform=v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)]),`
- **Line 38**: `target_transform=Lambda(lambda y: torch.zeros(10, dtype=torch.float).scatter_(0, torch.tensor(y), value=1))` -> `target_transform=v2.Lambda(lambda y: torch.nn.functional.one_hot(torch.tensor(y), num_classes=10).float())`
- **Lines 42-48**: Update the `ToTensor()` section heading and explanation to describe `v2.ToImage()` + `v2.ToDtype()`
- **Lines 60-61**: `target_transform = Lambda(lambda y: torch.zeros(10, dtype=torch.float).scatter_(dim=0, index=torch.tensor(y), value=1))` -> `target_transform = v2.Lambda(lambda y: torch.nn.functional.one_hot(torch.tensor(y), num_classes=10).float())`

## Files

- `beginner_source/basics/transforms_tutorial.py`

---
*Copied from https://github.com/sekyondaMeta/tutorials/issues/71*

cc @subramen

### PR DESCRIPTION
**Pull request by omkar-334**
Fixes #3853

## Description
files changed  - `beginner_source/basics/transforms_tutorial.py`

1. Replaced deprecated `torchvision.transforms.ToTensor` (deprecated since torchvision 0.16) with the v2 pipeline `v2.Compose([v2.ToImage(),  v2.ToDtype(torch.float32, scale=True)])`.
2. Switched the legacy v1 `transforms` namespace import to `from torchvision.transforms import v2`.
3.  Replaced the low-level `torch.zeros(...).scatter_(...)` one-hot pattern with `torch.nn.functional.one_hot(torch.tensor(y),
  num_classes=10).float()`, wrapped in `v2.Lambda`.
4.  Updated the section heading and explanatory prose so they describe `v2.ToImage()` + `v2.ToDtype()` and `F.one_hot` instead of `ToTensor`  and `scatter_`.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @subramen

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
@claude

**Comment by sekyondaMeta** [comment]
@omkar-334 Generally looks good, but can you fix the nits mentioned in the review above:

Line 38 is long (~100 chars) — Consider breaking the target_transform lambda across lines for readability, though this is a style preference and the current form is acceptable for a tutorial:

target_transform=v2.Lambda(
    lambda y: torch.nn.functional.one_hot(torch.tensor(y), num_classes=10).float()
)
Same applies to line 61 where it's repeated.

import torch.nn.functional as F — The PR uses torch.nn.functional.one_hot inline (twice). A common pattern in PyTorch tutorials is import torch.nn.functional as F and then F.one_hot(...). This would shorten the lambda and match conventions used elsewhere in PyTorch docs. However, since this tutorial is minimal and doesn't use F for anything else, the fully qualified name is also fine.

Further Reading link — The "Further Reading" section at line 70 still links to the general transforms page. Consider adding or replacing with a direct link to the [v2 transforms getting started guide](https://pytorch.org/vision/stable/auto_examples/transforms/plot_transforms_getting_started.html) for readers who want to explore the new API further.

**Comment by omkar-334** [comment]
hi @sekyondaMeta thanks for the review. i've applied the concerned changes and updated the branch...

**Review by sekyondaMeta — APPROVED** [review]
